From 7b147c8ce39ef964cf89b2dead4582301e863b5e Mon Sep 17 00:00:00 2001
From: siddhesh <siddhesh>
Date: Tue, 25 Sep 2012 12:20:35 +0000
Subject: [PATCH] gdb/ChangeLog:

	* c-typeprint.c (c_type_print_varspec_suffix): Remove cast and
	use plongest to print the array size.

testsuite/ChangeLog:

	* gdb.base/longest-types.c: New test case.
	* gdb.base/longest-types.exp: New test case.
---
 gdb/ChangeLog                            |    5 +++++
 gdb/c-typeprint.c                        |    4 ++--
 gdb/testsuite/ChangeLog                  |    5 +++++
 gdb/testsuite/gdb.base/longest-types.c   |   28 ++++++++++++++++++++++++++++
 gdb/testsuite/gdb.base/longest-types.exp |   25 +++++++++++++++++++++++++
 5 files changed, 65 insertions(+), 2 deletions(-)
 create mode 100644 gdb/testsuite/gdb.base/longest-types.c
 create mode 100644 gdb/testsuite/gdb.base/longest-types.exp

Index: gdb-7.2/gdb/testsuite/gdb.base/longest-types.c
===================================================================
--- /dev/null
+++ gdb-7.2/gdb/testsuite/gdb.base/longest-types.c
@@ -0,0 +1,28 @@
+/* This testcase is part of GDB, the GNU debugger.
+
+   Copyright 2012 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
+
+struct foo
+{
+  char buf[0xffff000000];
+  char buf2[2];
+} *f;
+
+int
+main (void)
+{
+  return 0;
+}
Index: gdb-7.2/gdb/testsuite/gdb.base/longest-types.exp
===================================================================
--- /dev/null
+++ gdb-7.2/gdb/testsuite/gdb.base/longest-types.exp
@@ -0,0 +1,26 @@
+# This testcase is part of GDB, the GNU debugger.
+
+# Copyright 2012 Free Software Foundation, Inc.
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+set testfile "longest-types"
+set srcfile ${testfile}.c
+
+if { [prepare_for_testing ${testfile}.exp ${testfile} $srcfile {debug quiet}] } {
+    return -1
+}
+
+# 64-bit array size should not overflow
+gdb_test "print &f->buf" {= \(char \(\*\)\[1099494850560\]\) 0x0}
